src/mbed_cloud/connect/connect.py

```diff
@@ -348,67 +349,55 @@ def set_resource_value_async(self, device_id, resource_path,
-        if fix_path and resource_path.startswith("/"):
-            resource_path = resource_path[1:]
+        resource_value_bytes = bytes(resource_value, encoding='utf-8')
+        payload_b64 = base64.b64encode(resource_value_bytes).decode('utf-8')
 
-        api = self._get_api(mds.ResourcesApi)
-        resp = api.update_resource_value(device_id,
-                                         resource_path,
-                                         resource_value)
-        return AsyncConsumer(resp.async_response_id, self._db)
+        return self._mds_rpc_post(
+            device_id,
+            method='PUT',
+            uri=resource_path,
+            content_type="text/plain",
+            payload_b64=payload_b64
+        )
 
     @catch_exceptions(mds.rest.ApiException)
-    def execute_resource(self, device_id, resource_path, fix_path=True, timeout=None, **kwargs):
+    def execute_resource(self, device_id, resource_path, fix_path=True, timeout=None):
         """Execute a function on a resource.
 
         Will block and wait for response to come through. Usage:
 
         .. code-block:: python
 
             try:
-                v = api.execute_resource(device, path, function_name)
+                v = api.execute_resource(device, path)
                 print("Success, returned value:", v)
             except AsyncError, e:
                 print("Error", e)
 
         :param str device_id: The name/id of the device (Required)
         :param str resource_path: The resource path to update (Required)
-        :param str resource_function: The function to trigger
-        :param fix_path: if True then the leading /, if found, will be stripped before
-            doing request to backend. This is a requirement for the API to work properly
+        :param str resource_function: Unused
+        :param fix_path: Unused
+        :param timeout: Timeout in seconds
         :raises: AsyncError
         :returns: The value returned from the function executed on the resource
         :rtype: str
         """
-        # Ensure we're listening to notifications first
         self.ensure_notifications_thread()
-
-        # When path starts with / we remove the slash, as the API can't handle //.
-        if fix_path and resource_path.startswith("/"):
-            resource_path = resource_path[1:]
-
-        api = self._get_api(mds.ResourcesApi)
-        resp = api.execute_or_create_resource(device_id,
-                                              resource_path,
-                                              **kwargs)
-        consumer = AsyncConsumer(resp.async_response_id, self._db)
-        return consumer.wait(timeout)
+        return self.execute_resource_async(device_id, resource_path).wait(timeout)
 
     @catch_exceptions(mds.rest.ApiException)
-    def execute_resource_async(self, device_id, resource_path, fix_path=True, **kwargs):
+    def execute_resource_async(self, device_id, resource_path):
```

Comment:
Removing the `fix_path` is definitely a breaking change, can you document it in the same way as you did for `set_resource_value`.

Removing the `kwargs` is more debatable as they were never documented.

Reply:
Good point. I've restored the fix_path parameter for both `execute_resource_async` and `set_resource_value_async`.